modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js

```diff
@@ -224,15 +224,40 @@ class CriteriaRow extends Component {
 	};
 
 	_handleTypedInputChange = (value, type) => {
-		const {criterion, onChange} = this.props;
-
-		onChange(
-			{
-				...criterion,
-				type,
-				value
-			}
-		);
+		const {criterion, index, onAdd, onChange} = this.props;
```

Comment:
I'd rather have the oData parser taking this decision.

Reply:
The most ideal way of displaying the selected items right now would be a new group reading something like `organization eq 'a' OR organization eq 'b'` and I thought it would be cleaner having data formatted in the same way sending it up to the odata parser.

I suppose it does make sense for this to be in the odata parser since this is a translation that we'll be doing converting an array to multiple rows.

I'll try handling this in the odata parser though. This could help expose the "custom" behavior and make it more ideal if we wanted to support `any` and `all` operators.

Reply:
I've come to the conclusion that having the check for array is better to be done in `CriteriaRow` because not only would we have to update the odata parser to handle arrays, but the criteria builder as well to keep the `criteriaMap` in sync with the query. It seems like more overhead managing the cases in multiple areas than directly creating new rows where the `criteriaMap` and `query` change consistently.

TLDR: I'm keeping the changes already made in this PR for the array handling.